packages/dd-trace/src/profiling/profilers/wall.js

```diff
@@ -286,14 +286,23 @@ class NativeWallProfiler {
 
     const labels = { ...getThreadLabels() }
 
-    const { context: { ref }, timestamp } = context
-    const { spanId, rootSpanId, webTags, endpoint } = ref ?? {}
-
     if (this._timelineEnabled) {
       // Incoming timestamps are in microseconds, we emit nanos.
-      labels[END_TIMESTAMP_LABEL] = timestamp * 1000n
+      labels[END_TIMESTAMP_LABEL] = context.timestamp * 1000n
+    }
+
+    const asyncId = context.asyncId
+    if (asyncId !== undefined && asyncId !== -1) {
+      labels['async id'] = asyncId
     }
 
+    const ref = context.context?.ref
```

Comment:
I guess a comment to explain why the context might be undefined would be useful in this case. I tried to think of even better ways like having a variable name for different context types or something the like, but I couldn't come up with anything great so far